jacobian_follower/src/jacobian_follower/jacobian_follower.cpp

```diff
@@ -175,105 +141,123 @@ PlanResult JacobianFollower::plan(planning_scene_monitor::LockedPlanningSceneRW
   PointSequence target_point_sequence;
   auto reached_target = true;
-  for (size_t waypoint_idx = 0; waypoint_idx < n_points; ++waypoint_idx)
-  {
-    for (auto const &gripper : grippers)
-    {
+  for (size_t waypoint_idx = 0; waypoint_idx < n_points; ++waypoint_idx) {
+    for (auto const &gripper : grippers) {
       target_point_sequence.emplace_back(gripper[waypoint_idx]);
     }
     EigenHelpers::VectorQuaterniond preferred_tool_orientations_q;
-    std::transform(preferred_tool_orientations.cbegin(),
-                   preferred_tool_orientations.cend(),
+    std::transform(preferred_tool_orientations.cbegin(), preferred_tool_orientations.cend(),
                    std::back_inserter(preferred_tool_orientations_q),
                    [](Eigen::Vector4d x) { return Eigen::Quaterniond{x}; });
-    auto const[traj, gripper_reached_target] = moveInWorldFrame(planning_scene,
-                                                                group_name,
-                                                                tool_names,
-                                                                preferred_tool_orientations_q,
-                                                                start_state,
-                                                                target_point_sequence);
+    auto const [traj, gripper_reached_target] = moveInWorldFrame(
+        planning_scene, group_name, tool_names, preferred_tool_orientations_q, start_state, target_point_sequence);
     reached_target = reached_target and gripper_reached_target;
 
     robot_trajectory.append(traj, 0);
   }
 
-  if (!time_param_.computeTimeStamps(robot_trajectory, max_velocity_scaling_factor, max_acceleration_scaling_factor))
-  {
+  if (!time_param_.computeTimeStamps(robot_trajectory, max_velocity_scaling_factor, max_acceleration_scaling_factor)) {
     ROS_ERROR_STREAM_NAMED(LOGGER_NAME, "Time parametrization for the solution path failed.");
   }
 
-  if (not robot_trajectory.empty())
-  {
+  if (not robot_trajectory.empty()) {
     visual_tools_.publishTrajectoryPath(robot_trajectory);
   }
 
   return {robot_trajectory, reached_target};
 }
 
-bool JacobianFollower::isRequestValid(std::string const &group_name,
-                                      std::vector<std::string> const &tool_names,
-                                      std::vector<std::vector<Eigen::Vector3d>> const &grippers) const
-{
-  if (not model_->hasJointModelGroup(group_name))
-  {
+std::vector<std::vector<double>> JacobianFollower::compute_IK_solutions(geometry_msgs::Pose target_pose,
+                                                                        const std::string &group_name) {
+  auto const jmg = model_->getJointModelGroup(group_name);
+  const kinematics::KinematicsBaseConstPtr &solver = jmg->getSolverInstance();
+
+  Eigen::Isometry3d solverTrobot = Eigen::Isometry3d::Identity();
+
+  robot_state::RobotStatePtr kinematic_state = std::make_shared<robot_state::RobotState>(model_);
```

Comment:
use auto here to prevent accidentally putting the mismatching types on the L and R sides